packages/vscode-dependencies-validation/test/diagnostics.spec.ts

```diff
@@ -0,0 +1,84 @@
+import proxyquire = require("proxyquire");
+import { diagnosticCollection } from "./vscodeMocks";
+import { createSandbox, SinonMock, SinonSandbox } from "sinon";
+import { refreshDiagnostics } from "../src/diagnostics";
+import { range } from "lodash";
+import { NPM_DEPENDENCY_ISSUES_CODE } from "../src/constants";
+
+describe("diagnostics unit test", () => {
+  const packageJsonPath = "/root/folder/package.json";
+  const Uri = {
+    file: (path: string) => `uri_${path}`,
+  };
+  class Range {
+    constructor(n1: number, n2: number, n3: number, n4: number) {}
+  }
+  const enum DiagnosticSeverity {
+    Error = 0,
+  }
+  class Diagnostic {
+    constructor(r: Range, m: string, ds: DiagnosticSeverity, code?: string) {}
+  }
+
+  let refreshDiagnosticsProxy: typeof refreshDiagnostics;
+  let problems: string[] | undefined;
+  let sandbox: SinonSandbox;
+  let diagnosticCollectionMock: SinonMock;
+
+  context("refreshDiagnostics()", () => {
+    before(() => {
+      sandbox = createSandbox();
+
+      const diagnosticsModule = proxyquire("../src/diagnostics", {
+        vscode: {
+          Range,
+          Diagnostic,
+          Uri,
+          "@noCallThru": true,
+        },
+        "@sap-devx/npm-dependencies-validation": {
+          findDependencyIssues: async () => Promise.resolve({ problems }),
+          "@noCallThru": true,
+        },
+      });
+
+      refreshDiagnosticsProxy = diagnosticsModule.refreshDiagnostics;
```

Comment:
when you init **refreshDiagnosticsProxy** only **once** in before
you make an assumption that ""../src/diagnostics"" is stateless.
This may be true now, but would it be true in the future too?
So might be safer to init it in **beforeEach**

Reply:
I changed it, but it case it will has state the tests will probably fail